datafeeder/src/main/java/org/georchestra/datafeeder/service/publish/impl/GeorchestraPublishingServicesConfiguration.java

```diff
@@ -32,10 +38,33 @@
  * @see DataFeederConfigurationProperties
  */
 @Configuration
-@Profile({ "!mock" })
+@Profile("!mock")
 public class GeorchestraPublishingServicesConfiguration {
 
     public @Bean DataBackendService dataBackendService() {
         return new GeorchestraDataBackendService();
     }
+
+    public @Bean OWSPublicationService owsPublicationService() {
+        return new GeorchestraOwsPublicationService();
+    }
+
+    public @Bean GeoServerClient geoServerApiClient(DataFeederConfigurationProperties props) {
+        ExternalApiConfiguration config = props.getPublishing().getGeoserver();
+        String restApiEntryPoint = config.getApiUrl().toExternalForm();
+
+        GeoServerClient client = new GeoServerClient(restApiEntryPoint);
+
+        Map<String, String> authHeaders = new HashMap<>();
+        // authHeaders.put("sec-proxy", "true");
+        authHeaders.put("sec-username", "datafeeder-application");
+        authHeaders.put("sec-roles", "ROLE_ADMINISTRATOR");
```

Comment:
Maybe we need a discussion with @pmauduit and others about the pros and cons of directly targeting the internal geoserver endpoint as opposed to going through the security-proxy via the front LB.
Usually we do the latter (eg: mapfishapp, extractorapp both have in their properties a reference to a `geoserver_privileged_user`), but I'm not opposed to changing the way we do if everyone agrees.
The downside of the current situation is that it is a potential security hole. The downside of targeting the geoserver local service is that the module configuration will depend on the implementation (localhost:8080/geoserver or geoserver:8080/geoserver).

Reply:
:+1: please no hardcoding of username/target server :)

Reply:
Hi, I came to see the comments with the idea in mind of fully externalizing the configuration for the auth method, so good timing :)
i.e., we currently have
```
publishing.geoserver.api-url=http://localhost:8080/geoserver/rest
publishing.geoserver.username=admin
publishing.geoserver.password=geoserver
```

My idea is to turn it into:

```
publishing.geoserver.api-url=http://localhost:8080/geoserver/rest
publishing.geoserver.auth.type=basic|headers

publishing.geoserver.auth.basic.username=admin
publishing.geoserver.auth.basic.password=geoserver

publishing.geoserver.auth.headers.sec-username=XXX
publishing.geoserver.auth.headers.sec-roles=YYY
publishing.geoserver.auth.headers.<keyM>=<valueN>
```

Sounds better?

Reply:
yes,  i think that would work for me (for historical reasons i dont use `geoserver_privileged_user` from the LDAP, and my geoserver has a local admin user not in the LDAP..)

thanks !

Reply:
See https://jira.camptocamp.com/browse/GSGEODPT43-139

Reply:
(Landry does not have access to JIRA - which is our internal bug tracker)

Reply:
and that's fine with me not accessing to such monstrosity... my last experience with geoserver's JIRA was .. painful, to say the least ;)

Reply:
Moved this discussion/decision to its own task. Merging in order to continue working.